blocks/library/button/index.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+import { registerBlock, query } from 'api';
+import Editable from 'components/editable';
+import IconButton from '../../../editor/components/icon-button';
+
+const { attr, children } = query;
+
+/**
+ * Returns an attribute setter with behavior that if the target value is
+ * already the assigned attribute value, it will be set to undefined.
+ *
+ * @param  {string}   align Alignment value
+ * @return {Function}       Attribute setter
+ */
+function applyOrUnset( align ) {
```

Comment:
Per @iseulde's comment at https://github.com/WordPress/gutenberg/issues/401#issuecomment-293732082, maybe there's value in allowing the control to be associated with an attribute to toggle, even just as an optional convenience.

Also, this has since been renamed in the image block as `toggleAlignment`.